src/Composer/JanusPlugin.php

```diff
@@ -0,0 +1,165 @@
+<?php declare(strict_types=1);
+
+namespace Janus\Composer;
+
+use Composer\Composer;
+use Composer\DependencyResolver\Operation\InstallOperation;
+use Composer\DependencyResolver\Operation\OperationInterface;
+use Composer\DependencyResolver\Operation\UpdateOperation;
+use Composer\EventDispatcher\EventSubscriberInterface;
+use Composer\Installer\PackageEvent;
+use Composer\Installer\PackageEvents;
+use Composer\IO\IOInterface;
+use Composer\Plugin\PluginInterface;
+use Composer\Script\Event;
+use Composer\Script\ScriptEvents;
+use Janus\Command\InitializeCommand;
+use Symfony\Component\Process\Process;
+
+/**
+ * @final
+ */
+class JanusPlugin implements PluginInterface, EventSubscriberInterface
+{
+	private bool $hadJanusOperation = false;
+
+	/**
+	 * @inheritDoc
+	 */
+	#[\Override]
+	public function activate (Composer $composer, IOInterface $io) : void {}
+
+	/**
+	 * @inheritDoc
+	 */
+	#[\Override]
+	public function deactivate (Composer $composer, IOInterface $io) : void
+	{
+		// nothing to do
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	#[\Override]
+	public function uninstall (Composer $composer, IOInterface $io) : void
+	{
+		// nothing to do
+	}
+
+	/**
+	 *
+	 */
+	public function checkForJanusOperations (PackageEvent $event) : void
+	{
+		foreach ($event->getOperations() as $operation)
+		{
+			if ($this->isJanusUpdate($operation))
+			{
+				$this->hadJanusOperation = true;
+				break;
+			}
+		}
+	}
+
+	/**
+	 *
+	 */
+	private function isJanusUpdate (OperationInterface $operation) : bool
+	{
+		if ($operation instanceof InstallOperation)
+		{
+			return "21torr/janus" === $operation->getPackage()->getName();
+		}
+
+		if ($operation instanceof UpdateOperation)
+		{
+			return "21torr/janus" === $operation->getTargetPackage()->getName();
+		}
+
+		return false;
+	}
+
+	/**
+	 * Callback after the autoloader was dumped
+	 */
+	public function afterAutoloadDump (Event $event) : void
+	{
+		if (!$this->hadJanusOperation)
+		{
+			return;
+		}
+
+		$this->hadJanusOperation = false;
+
+		$io = $event->getIO();
+		$io->write("\n<fg=magenta>Janus update detected, running janus update</>\n");
+
+		$selected = $io->select(
+			"What are you currently using?",
+			InitializeCommand::ALLOWED_TYPES,
+			"library",
+		);
+		$type = InitializeCommand::ALLOWED_TYPES[$selected] ?? null;
+
+		$vendorDir = $event->getComposer()->getConfig()->get('vendor-dir');
+		\assert(\is_string($vendorDir));
+
+		$success = $this->runJanus($io, $vendorDir, $type);
+
+		if ($success)
+		{
+			$io->write("\n<fg=green>Janus installation complete.</>\n");
+		}
+		else
+		{
+			$io->writeError("\n<fg=red>Janus installation failed, please run it manually: `composer exec janus init`</>\n");
+		}
+	}
+
+	/**
+	 * Runs Janus
+	 */
+	private function runJanus (
+		IOInterface $io,
+		string $vendorDir,
+		?string $type,
+	) : bool
+	{
+		$command = [
+			"{$vendorDir}/bin/janus",
```

Comment:
Does this break when we use Janus in a `vendor-bin` directory?

Reply:
Is this something that we'd want to address, or should we rather update our docs, as we currently explicitly list it as a workaround/option?